File: Baekjoon/유석환/Q7576.swift
let MN = readLine()!.split(separator: " ").map { Int(String($0))! }
let M = MN[0]
let N = MN[1]
var graph = [[Int]]()

// 그래프 입력받기
for _ in 0..<N {
    graph.append(readLine()!.split(separator: " ").map { Int(String($0))! })
}

// 토마토 좌표 배열
var tomatoes = [(Int, Int)]()
for i in 0..<M {
    for j in 0..<N {
        if graph[j][i] == 1 {
            tomatoes.append((i, j))
        }
    }
}

// 모두 익을 때까지의 최소 일수 계산
var day = -1
while !tomatoes.isEmpty {
    // 인접 토마토 배열
    var adjacentTomatoes = [(Int, Int)]()
    
    // 인접 토마토 익히기
    for tomato in tomatoes {
        // 상
        if tomato.1 - 1 >= 0 && graph[tomato.1 - 1][tomato.0] == 0 {
            graph[tomato.1 - 1][tomato.0] = 1
            adjacentTomatoes.append((tomato.0, tomato.1 - 1))
        }
        
        // 하
        if tomato.1 + 1 <= N - 1 && graph[tomato.1 + 1][tomato.0] == 0 {
            graph[tomato.1 + 1][tomato.0] = 1
            adjacentTomatoes.append((tomato.0, tomato.1 + 1))
        }
        
        // 좌
        if tomato.0 - 1 >= 0 && graph[tomato.1][tomato.0 - 1] == 0 {
            graph[tomato.1][tomato.0 - 1] = 1
            adjacentTomatoes.append((tomato.0 - 1, tomato.1))
        }
        
        // 우
        if tomato.0 + 1 <= M - 1 && graph[tomato.1][tomato.0 + 1] == 0 {
            graph[tomato.1][tomato.0 + 1] = 1
            adjacentTomatoes.append((tomato.0 + 1, tomato.1))
        }
    }
    
    tomatoes = adjacentTomatoes
    day += 1
}

// 모두 익지 못하는 상황이면 day를 -1로 수정
outer: for i in 0..<M {
    for j in 0..<N {
        if graph[j][i] == 0 {
            day = -1
            break outer
        }
    }
}

// 정답 출력
print(day)

File: Baekjoon/김경훈/Q7576.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input.shift().split(' ').map(Number);

let graph = [];

const dX = [1, -1, 0, 0];
const dY = [0, 0, 1, -1];

let total = N * M;
let ripedTomato = [];
let newRipedTomato = [];

for(line of input) {
    graph.push(line.split(' ').map(Number));
}

for(let i = 0; i < M; i++) {
    for(let j = 0; j < N; j++) {
        if(graph[i][j] === 1) {
            ripedTomato.push(`${i} ${j}`);
        } else if(graph[i][j] === -1) {
            total -= 1;
        } 
    }
}

let day = 0;

let tomatoCount = ripedTomato.length;
while (true) {
    ripedTomato.forEach((tomato) => {
        const [n, m] = tomato.split(' ').map(Number);
        for(let i = 0; i < 4; i++) {
            const nextX = m + dX[i];
            const nextY = n + dY[i];
            if(nextY >= 0 && nextY < M && nextX >= 0 && nextX < N) {
                if(graph[nextY][nextX] === 0) {
                    graph[nextY][nextX] = 1;
                    newRipedTomato.push(`${nextY} ${nextX}`);
                } else { // 빈칸, 이미 익은 토마토 continue
                    continue;
                }
            }
        }

    });

    if(newRipedTomato.length === 0) { // 새로 익은 토마토가 없으면 break
        break;
    }
    
    tomatoCount += newRipedTomato.length; // 카운트 세줌
    ripedTomato = Array.from(newRipedTomato); // 새로 익은 토마토로 최신화
    newRipedTomato = [];
    day += 1;

}

console.log(tomatoCount === total ? day : '-1'); // 빈칸을 제외한 total과 익은 토마토 개수를 카운트 한 것을 비교